### ISSUE TITLE
Can't start on node v7.10.1, Mac Os 10.12.6

### ISSUE THREAD

**Issue by anija** [slanatech/swagger-stats#22]
Hi,

i'm trying to install swagger stats but i get this error when i start the app:
```
/src/node_modules/prom-client/lib/gauge.js:219
                        throw new TypeError(Value is not a valid number: ${util.format(value)});
                        ^

TypeError: Value is not a valid number: undefined
    at /src/node_modules/prom-client/lib/gauge.js:219:10
    at Gauge.set (/src/node_modules/prom-client/lib/gauge.js:85:31)
    at swsCoreStats.calculateSystemStats (/src/node_modules/swagger-stats/lib/swsCoreStats.js:151:65)
    at swsCoreStats.tick (/src/node_modules/swagger-stats/lib/swsCoreStats.js:121:10)
    at Timeout.swsProcessor.tick (/src/node_modules/swagger-stats/lib/swsProcessor.js:137:20)
    at Timeout.ontimeout [as _repeat] (timers.js:400:34)
    at Timeout.wrapper [as _onTimeout] (timers.js:425:11)
    at tryOnTimeout (timers.js:232:11)
    at Timer.listOnTimeout (timers.js:202:5)
```

This is the line #151 of swsCoreStats.js:
`this.promClientMetrics.nodejs_process_memory_external_bytes.set(this.sys.external);`

This is this.sys content:
```
  rss: 95752192,
  heapTotal: 75632640,
  heapUsed: 46559352,
  external: undefined,
  cpu: 26.9837199868224
```
And this is the result of process.memoryUsage():
```
{ rss: 95752192, heapTotal: 75632640, heapUsed: 46559352 }
```

The external property is missing.

**Comment by sv2** [comment]
Thank you! will address shortly

**Comment by reuben453** [comment]
Hi,

I'm facing a very similar (or exactly same) issue on node v6.2.1, mac v10.12.6.

This is the error I get when I start the app:
```
/Users/User/projects/nu-metrics/node_modules/prom-client/lib/gauge.js:219
			throw new TypeError(`Value is not a valid number: ${util.format(value)}`);
			^

TypeError: Value is not a valid number: undefined
    at /Users/User/projects/nu-metrics/node_modules/prom-client/lib/gauge.js:219:10
    at Gauge.set (/Users/User/projects/nu-metrics/node_modules/prom-client/lib/gauge.js:85:31)
    at swsCoreStats.calculateSystemStats (/Users/User/projects/nu-metrics/node_modules/swagger-stats/lib/swsCoreStats.js:151:65)
    at swsCoreStats.tick (/Users/User/projects/nu-metrics/node_modules/swagger-stats/lib/swsCoreStats.js:121:10)
    at Timeout.swsProcessor.tick (/Users/User/projects/nu-metrics/node_modules/swagger-stats/lib/swsProcessor.js:137:20)
    at Timeout.ontimeout [as _repeat] (timers.js:392:34)
    at Timeout.wrapper [as _onTimeout] (timers.js:417:11)
    at tryOnTimeout (timers.js:224:11)
    at Timer.listOnTimeout (timers.js:198:5)
```

Should I create a separate issue?

**Comment by reuben453** [comment]
Line 151 of `nu-metrics/node_modules/swagger-stats/lib/swsCoreStats.js:151` is:
`this.promClientMetrics.nodejs_process_memory_external_bytes.set(this.sys.external);`

I checked the this.sys content : 
```
this.sys { rss: 115826688,
  heapTotal: 93437952,
  heapUsed: 56715592,
  external: undefined,
  cpu: 11.12675279025026 }
```

**Comment by sv2** [comment]
@reuben453, yes, I believe it's the same issue. "external" is missing in process.memoryUsage() result; I've added fix to handle this properly, and will release version with this fix shortly. Thanks!

**Comment by sv2** [comment]
addressed in 0.93.1

**Comment by anija** [comment]
Thank you!
